lib/types/result-set.js

```diff
@@ -10,30 +10,76 @@ const asyncIteratorSymbol = Symbol.asyncIterator || "@@asyncIterator";
 
 /** @module types */
 
-// Old definition of the Result set
-// /**
-//  * Creates a new instance of ResultSet.
-//  * @class
-//  * @classdesc Represents the result of a query.
-//  * @param {Object} response
-//  * @param {String} host
-//  * @param {Object} triedHosts
-//  * @param {Number} speculativeExecutions
-//  * @param {Number} consistency
-//  * @param {Boolean} isSchemaInAgreement
-//  * @constructor
-//  */
-// function ResultSet(response, host, triedHosts, speculativeExecutions, consistency, isSchemaInAgreement) {
-
 /**
  * Represents the result of a query.
  */
 
 class ResultSet {
+    // Old definition of the Result set
+    // /**
+    //  * @param {Object} result
+    //  * @param {String} host
+    //  * @param {Object} triedHosts
+    //  * @param {Number} speculativeExecutions
+    //  * @param {Number} consistency
+    //  * @param {Boolean} isSchemaInAgreement
+    //  */
     /**
      * @param {rust.QueryResultWrapper} result
      */
-    constructor(result) {
+    constructor(
+        result,
+        host,
+        triedHosts,
+        speculativeExecutions,
+        consistency,
+        isSchemaInAgreement,
+    ) {
+        if (!(result instanceof rust.QueryResultWrapper)) {
```

Comment:
So you say it's an old constructor (with old code) simply re-introduced?

Reply:
wrapped in the if statement with removed comments (those comments are in the "new" path of the constructor)